extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusSecurityIdentity.java

```diff
@@ -239,4 +240,91 @@ public QuarkusSecurityIdentity build() {
             return new QuarkusSecurityIdentity(this);
         }
     }
+
+    /**
+     * This map only allows adding new entries
+     */
+    static final class AttributesMap implements Map<String, Object> {
```

Comment:
The identity was designed to be immutable. I don't know if we want to change this.

Reply:
@stuartwdouglas sure but augmentors can change the identity by adding new attributes,  so This is a workaround to be able to do the same in HttpAuthenticator, the existing attributes can't be affected.  
I don't mind try something else but it seems like the only option to make all identities made visible to CDI producers for them to fi d the right one

Reply:
@stuartwdouglas oh yes, see your suggestion below,  thanks